crypto/dsa/dsa.go

```diff
@@ -0,0 +1,87 @@
+package dsa
+
+import (
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/crypto/dsa/ecdsa"
+	"github.com/tbd54566975/web5-go/crypto/dsa/eddsa"
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+var AlgorithmID = struct {
+	SECP256K1 string
+	ED25519   string
+}{
+	SECP256K1: ecdsa.SECP256K1AlgorithmID,
+	ED25519:   eddsa.ED25519AlgorithmID,
+}
+
+func GeneratePrivateKey(algorithmID string) (jwk.JWK, error) {
+
+	var privateKey jwk.JWK
+	var err error
+
+	if ecdsa.SupportsAlgorithmID(algorithmID) {
+		privateKey, err = ecdsa.GeneratePrivateKey(algorithmID)
+	} else if eddsa.SupportsAlgorithmID(algorithmID) {
+		privateKey, err = eddsa.GeneratePrivateKey(algorithmID)
+	} else {
+		err = fmt.Errorf("unsupported algorithm: %s", algorithmID)
+	}
+
+	return privateKey, err
+}
+
+func GetPublicKey(privateKey jwk.JWK) jwk.JWK {
+	switch privateKey.KTY {
+	case ecdsa.KeyType:
+		return ecdsa.GetPublicKey(privateKey)
+	case eddsa.KeyType:
+		return eddsa.GetPublicKey(privateKey)
+	default:
+		return jwk.JWK{}
+	}
+}
+
+func Sign(payload []byte, jwk jwk.JWK) ([]byte, error) {
+	var err error
+	var signature []byte
+
+	switch jwk.KTY {
+	case ecdsa.KeyType:
+		signature, err = ecdsa.Sign(payload, jwk)
+	case eddsa.KeyType:
+		signature, err = eddsa.Sign(payload, jwk)
+	default:
+		err = fmt.Errorf("unsupported key type: %s", jwk.KTY)
+	}
+
+	return signature, err
+}
```

Comment:
You typically would just return directly in the switch:

```go
func Sign(payload []byte, jwk jwk.JWK) ([]byte, error) {
	switch jwk.KTY {
	case ecdsa.KeyType:
		return ecdsa.Sign(payload, jwk)
	case eddsa.KeyType:
		return eddsa.Sign(payload, jwk)
	default:
		return nil, fmt.Errorf("unsupported key type: %s", jwk.KTY)
	}
}
```

It removes a lot of noise and will make the code much more readable.

Reply:
done: https://github.com/TBD54566975/web5-go/commit/633975960e39e1490643d9fcca30497ef8e10491